presto-hive/src/main/java/com/facebook/presto/hive/parquet/ParquetHiveRecordCursor.java

```diff
@@ -203,6 +201,7 @@ else if (type.equals(DATE)) {
             }
         }
 
+        this.dataSource = buildHdfsParquetDataSource(path, configuration, start, length);
```

Comment:
Please don't forget to close this. This may leak critical resources if you are using the S3 filesystem.